src/Microsoft.AspNet.Mvc.Razor/RazorPage.cs

```diff
@@ -617,6 +614,20 @@ public void EndContext()
             PageExecutionContext?.EndContext();
         }
 
+        /// <summary>
+        /// Sets anti-forgery cookie and X-Frame-Options header on the response.
+        /// </summary>
+        /// <returns>A <see cref="HtmlString"/> that returns a <see cref="HtmlString.Empty"/>.</returns>
+        /// <remarks> Call this method to send anti-forgery cookie token and X-Frame-Options header to client
+        /// before <see cref="FlushAsync"/> flushes the headers. </remarks>
+        public virtual HtmlString SetAntiForgeryCookieAndHeader()
+        {
+            var antiForgery = Context?.RequestServices.GetRequiredService<AntiForgery>();
```

Comment:
will the context ever be null? if not remove ?. with .  in both the occurrences in this code.